Feature/41 message processor module impl

p-borozdin opened the pull request.

p-borozdin committed:
Добавил модуль message-storage-api.

p-borozdin committed:
refs #41 Добавили модуль message-storage-api и StorableMessage.

p-borozdin committed:
Добавил StorableMessage и изменил API интерфейса MessageStorage на работу с StorableMessage.

Commit:
refs #45 Исправляет ошибку в модуле регистрации пользователей

p-borozdin committed:
refs #41 Добавили поле String messageId в StorableMessage и реализовали сохранение сообщения в базе.

p-borozdin committed:
Вставил использование переменной toNockname ри создании StorableMessage в методе MessageProcessorImpl.submit()

ArtemGemuzov committed:
refs #41 Добавил аннотацию @Async к методу submit

ArtemGemuzov committed:
Merge remote-tracking branch 'origin/feature/41-message-processor-module-impl' into feature/41-message-processor-module-impl

p-borozdin committed:
Добавил модуль common с константами.

p-borozdin committed:
Добавил обратный ответ в случае ошибки (MessageProcessorImpl.sendErrorBackToTalker()).

avtseben commented on message-processor-impl/src/main/java/ru/jafti/braintalk/message/processor/impl/MessageProcessorImpl.java:
```diff
@@ -60,4 +85,23 @@ private void sendToOnlineTalker(TalkersMessage talkersMessage, UUID toTalkerGuid
+        if (messageChannel.isOnline(senderId)) {
+            SendMessageRequest messageRequest = new SendMessageRequest(
+                    talkersMessage.from().nickname(),
+                    CommonConstants.SYSTEM_TALKER,
```
Зачем отправлять сообщение системному боту?

Нужно было отправить обратно человеку сообщение от системного бота с текстом ошибки

ArtemGemuzov replied:
Первый никнейм указывает кому отправить сообщение, а второй - от кого. Соответственно сообщение идет от системы к человеку.

avtseben replied:
Давайте сделаем по другому, отдельный метод и отдельный DTO для отправки сигнальных сообщений 
SendMessageRequest предназначен для отправки сообщений от пользователя к пользователю.

Вот такой контракт 
```
public record SignalMessage(
        To to,
        String signalText
) {
    public record To(UUID talkerGuid) {
    }
}
```

avtseben replied:
```
public interface OnlineMessageChannel {
    void signal(SignalMessage message);
```

ArtemGemuzov replied:
Добавил отдельный метод и DTO

ArtemGemuzov committed:
#41 Исправление ошибок

  1. Добавил аннотацию @EnableAsync классу BrainTalkServerApplication
  2. Убрал дублирование кода в классе MessageProcessorImpl
  3. Поменял сообщение, отправляемое в случае ошибки
  4. Перевел контроллеры на использование SYSTEM_TALKER ("SystemBot") из класса CommonConstants

ArtemGemuzov committed:
#41 Изменения в обработке ошибок

  1. Добавил record SignalMessage для сообщений об ошибке
  2. Добавил метод signal(SignalMessage) для интерфейса OnlineMessageChannel
  3. Добавил реализацию signal(SignalMessage) для класса RendezvousPoint
  4. Теперь MessageProcessorImpl использует новые классы и методы для сообщения об ошибках
